docs/troubleshooting.md

```diff
@@ -18,6 +18,14 @@ If the language stats bar is reporting a language that you don't expect:
 Keep in mind that the repository language stats are only [updated when you push changes](how-linguist-works.md#how-linguist-works-on-githubcom), and the results are cached for the lifetime of your repository.
 If you have not made any changes to your repository in a while, you may find pushing another change will correct the stats.
 
+## When I click on a language present in the language stats bar, I get a message saying "Your search did not match any code"
+
+There are a few reasons that could lead to this outcome:
+
+1. If you are making an override via the `linguist-language` attribute, it will not be taken into account in GitHub's search results since GitHub Search relies on [go-enry](https://github.com/go-enry/go-enry) which doesn't support overrides at the moment ([More info](https://github.com/src-d/enry/issues/18)).
+2. It could be due to the fact that go-enry isn't using the latest version of Linguist which means that files could be detected differently in search compared to what Linguist detects for the status bar (see also the note at the end of [this section](#my-linguist-pr-has-been-merged-but-github-doesnt-reflect-my-changes)).
+3. Rarely, it can be that the files are associated to a language that is part of a group, this means that they will be counted as the parent language in the language stats bar, but as the actual language in search. For instance, a file ending with `.f90`, it is considered to be a "Fortran" in the stats bar, but "Fortran Free Form" in search.
+
```

Comment:
There’s also a fourth explanation: the [code search limitations](https://docs.github.com/en/search-github/github-code-search/about-github-code-search#limitations). I thought we had a reference to these somewhere but I can’t find it now (AFK on mobile only) so I think we should mention them here too.

Reply:
Good to know! I've added that to the list.